mempool/mempool_test.go

```diff
@@ -971,6 +971,10 @@ func TestVoteOrphan(t *testing.T) {
 		t.Fatalf("ProcessTransaction: accepted transaction references " +
 			"outputs of unknown or fully-spent transaction")
 	}
+	if code, ok := extractErrorCode(err); !ok || code != ErrOrphan {
```

Comment:
I believe this, and all the other cases below, can just replace the `if err == nil` case since if the error is nil it certainly won't be the expected code.

Then, if combined with the suggest `IsErrorCode` function, these can all turn into the, what is in my opinion more readable:

```Go
if !IsErrorCode(err, ErrOrphan) {
	t.Fatalf(....)
}
```